content/build/checks.md

```diff
@@ -0,0 +1,203 @@
+---
+title: Checking your build configuration
+description: Learn how to use build checks to validate your build configuration.
+keywords: build, buildx, buildkit, checks, validate, configuration, lint
+---
+
+{{< introduced buildx 0.15.0 >}}
+
+Build checks are a feature introduced in Dockerfile 1.8. It lets you validate
+your build configuration and conduct a series of checks prior to executing your
+build. Think of it as an advanced form of linting for your Dockerfile and build
+options, or a dry-run mode for builds.
+
+You can find the list of checks available, and a description of each, in the
+[Build checks reference](/reference/build-checks/).
+
+## How build checks work
+
+Typically, when you run a build, Docker executes the build steps in your
+Dockerfile and build options as specified. With build checks, rather than
+executing the build steps, Docker checks the Dockerfile and options you provide
+and reports any issues it detects.
+
+Build checks are useful for:
+
+- Validating your Dockerfile and build options before running a build.
+- Ensuring that your Dockerfile and build options are up-to-date with the
+  latest best practices.
+- Identifying potential issues or anti-patterns in your Dockerfile and build
+  options.
+
+## Build with checks
+
+Build checks are supported in Buildx version 0.15.0 and later. Invoking a build
+runs the checks by default, and displays any violations in the build output.
+For example, the following command both builds the image and runs the checks:
+
+```console
+$ docker build .
+[+] Building 3.5s (11/11) FINISHED
+...
+
+1 warning found (use --debug to expand):
+  - Lint Rule 'JSONArgsRecommended': JSON arguments recommended for CMD to prevent unintended behavior related to OS signals (line 7)
+
+```
+
+In this example, the build ran successfully, but a
+[JSONArgsRecommended](/reference/build-checks/json-args-recommended/) warning
+was reported, because `CMD` instructions should use JSON array syntax.
+
+### More verbose output
+
+Check warnings for a regular `docker build` display a concise message
+containing the rule name, the message, and the line number of where in the
+Dockerfile the issue originated. If you want to see more detailed information
+about the checks, you can use the `--debug` flag. For example:
+
+```console
+$ docker build --debug .
+[+] Building 3.5s (11/11) FINISHED
+...
+
+JSONArgsRecommended - https://docs.docker.com/go/dockerfile/rule/json-args-recommended/
```

Comment:
Per conversation with @colinhemmings, [#5037](https://github.com/moby/buildkit/pull/5037) is set to update the format to include a `WARNING` prefix, so the examples might need to be updated.